Primitive beans in factory cause build failure

When you try to register a primitive bean in a factory e.g.:
```java
@Factory
public class AFactory {
  @Bean
  @Singleton
  @Named("coolNumber")
  public int coolNumber() {
    return 15;
  }
}
```
The following error happens:
```
C:\...\AFactory$DI.java:27: error: illegal start of expression
    if (builder.isAddBeanFor("coolnumber", )) {
```
Generated code:
```java
public static void build_coolNumber(Builder builder) {
  if (builder.isAddBeanFor("coolnumber", )) {
    var factory = builder.get(AFactory.class);
    var bean = factory.coolNumber();
    builder.register(bean);
  }
}
```

--- a/inject-generator/src/main/java/io/avaje/inject/generator/PrimitiveUtil.java
+++ b/inject-generator/src/main/java/io/avaje/inject/generator/PrimitiveUtil.java
@@ -1,0 +1,24 @@
+package io.avaje.inject.generator;
+
+import java.util.Map;
+
+public final class PrimitiveUtil {
+
+  private PrimitiveUtil() {}
+
+  static final Map<String, String> wrapperMap =
+      Map.of(
+          "char", "Character",
+          "byte", "Byte",
+          "int", "Integer",
+          "long", "Long",
+          "short", "Short",
+          "double", "Double",
+          "float", "Float",
+          "boolean", "Boolean");
+
+  public static String wrap(String shortName) {
+    final var wrapped = wrapperMap.get(shortName);
+    return wrapped != null ? "java.lang." + wrapped : null;
+  }
+}

--- a/inject-generator/src/main/java/io/avaje/inject/generator/ProcessingContext.java
+++ b/inject-generator/src/main/java/io/avaje/inject/generator/ProcessingContext.java
@@ -136,7 +136,10 @@ final class ProcessingContext {
   }
 
   Element asElement(TypeMirror returnType) {
-    return typeUtils.asElement(returnType);
+
+    var wrapper = PrimitiveUtil.wrap(returnType.toString());
+    
+    return wrapper == null ? typeUtils.asElement(returnType) : element(wrapper);
   }
 
   void addModule(String moduleFullName) {

--- a/inject-generator/src/main/java/io/avaje/inject/generator/TypeExtendsReader.java
+++ b/inject-generator/src/main/java/io/avaje/inject/generator/TypeExtendsReader.java
@@ -186,9 +186,7 @@ final class TypeExtendsReader {
   private void readInterfacesOf(TypeMirror anInterface) {
     String rawType = Util.unwrapProvider(anInterface.toString());
     if (JAVA_LANG_OBJECT.equals(rawType)) {
-      return;
-    }
-    if (rawType.indexOf('.') == -1) {
+    } else if (rawType.indexOf('.') == -1) {
       context.logWarn("skip when no package on interface " + rawType);
     } else if (Constants.AUTO_CLOSEABLE.equals(rawType) || Constants.IO_CLOSEABLE.equals(rawType)) {
       closeable = true;
@@ -202,8 +200,11 @@ final class TypeExtendsReader {
         }
       }
       interfaceTypes.add(rawType);
-      for (TypeMirror supertype : context.types().directSupertypes(anInterface)) {
-        readInterfacesOf(supertype);
+      if (!rawType.startsWith("java.lang.")) {
+
+        for (TypeMirror supertype : context.types().directSupertypes(anInterface)) {
+          readInterfacesOf(supertype);
+        }
       }
     }
   }

--- a/inject-generator/src/main/java/io/avaje/inject/generator/package-info.java
+++ b/inject-generator/src/main/java/io/avaje/inject/generator/package-info.java
@@ -1,21 +1,19 @@
-@io.avaje.prism.GeneratePrisms({
-  @GeneratePrism(value = InjectModule.class),
-  @GeneratePrism(value = Factory.class),
-  @GeneratePrism(value = Singleton.class),
-  @GeneratePrism(value = Component.class),
-  @GeneratePrism(value = Prototype.class),
-  @GeneratePrism(value = Scope.class),
-  @GeneratePrism(value = Qualifier.class),
-  @GeneratePrism(value = Named.class),
-  @GeneratePrism(value = Inject.class),
-  @GeneratePrism(value = Aspect.class),
-  @GeneratePrism(value = Primary.class),
-  @GeneratePrism(value = Secondary.class),
-  @GeneratePrism(value = Proxy.class),
-  @GeneratePrism(value = DependencyMeta.class),
-  @GeneratePrism(value = Bean.class),
-  @GeneratePrism(value = io.avaje.inject.spi.Generated.class),
-})
+@GeneratePrism(InjectModule.class)
+@GeneratePrism(Factory.class)
+@GeneratePrism(Singleton.class)
+@GeneratePrism(Component.class)
+@GeneratePrism(Prototype.class)
+@GeneratePrism(Scope.class)
+@GeneratePrism(Qualifier.class)
+@GeneratePrism(Named.class)
+@GeneratePrism(Inject.class)
+@GeneratePrism(Aspect.class)
+@GeneratePrism(Primary.class)
+@GeneratePrism(Secondary.class)
+@GeneratePrism(Proxy.class)
+@GeneratePrism(DependencyMeta.class)
+@GeneratePrism(Bean.class)
+@GeneratePrism(Generated.class)
 package io.avaje.inject.generator;
 
 import io.avaje.inject.Bean;
@@ -27,6 +25,7 @@ import io.avaje.inject.Prototype;
 import io.avaje.inject.Secondary;
 import io.avaje.inject.aop.Aspect;
 import io.avaje.inject.spi.DependencyMeta;
+import io.avaje.inject.spi.Generated;
 import io.avaje.inject.spi.Proxy;
 import io.avaje.prism.GeneratePrism;
 import jakarta.inject.Inject;

